ast/compile.go

```diff
@@ -400,12 +415,21 @@ func (c *Compiler) Compile(modules map[string]*Module) {
 	c.init()
 
 	c.Modules = make(map[string]*Module, len(modules))
+	c.sorted = make([]string, 0, len(modules))
+	c.modules = nil
 
 	for k, v := range modules {
 		c.Modules[k] = v.Copy()
 		c.sorted = append(c.sorted, k)
 	}
 
+	if c.keepModules {
+		c.modules = make(map[string]*Module, len(modules))
+		for k, v := range modules {
+			c.modules[k] = v.Copy()
```

Comment:
❔ The question is whether or not we need the Copy here. Perhaps not?

Reply:
Nit: This could be part of the above loop I guess.

Reply:
yeah I had that first but then separated it... didn't want two `if c.keepModules` 🙃

Reply:
I'll drop the Copy tomorrow.